modules/ROOT/pages/java-support.adoc

```diff
@@ -169,6 +169,27 @@ If you are a MuleSoft customer:
 . Test your connector using xref:customer-connector-upgrade.adoc#test-your-custom-connector-with-munit[MUnit].
 . xref:mule-runtime::deploying.adoc[Deploy your applications] to a target running on Mule 4.6 or later.
 
+[[custom-code-in-connectors]]
+== Custom Code in Connectors 
+
+If you have custom code in your connectors, or if you are using third-party libraries, you must update that custom code to be compatible with Java 17.
+
+* Reflection on Java classes
++
+To ensure compatibility with Java 17, review and update any code that uses reflection to access Java classes or set parameter values. Java 17 imposes stricter access controls, so you should avoid introspecting JDK internals. Ensure you have a robust test suite and make sure it is run with Java 17. Pay special attention to the `setAccessible(true)` method because these type of changes in the accessibility is not allowed in Java 17.
```

Comment:
```suggestion
To ensure compatibility with Java 17, review and update any code that uses reflection to access Java classes or set parameter values. Java 17 imposes stricter access controls, so avoid introspecting JDK internals. Have a robust test suite that runs with Java 17. Pay special attention to the `setAccessible(true)` method because these types of changes in the accessibility are not allowed in Java 17.
```
I'm not clear what "these types of changes" refers to here in relation to setAccessible(true)